compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/AbstractBeginNode.java

```diff
@@ -116,6 +118,23 @@ public void generate(NodeLIRBuilderTool gen) {
         };
     }
 
+    public List<Node> getBlockNodeList() {
+        StructuredGraph.ScheduleResult schedule = graph().getLastSchedule();
+        if (schedule == null) {
+            return null;
+        }
+
+        Block[] blocks = schedule.getCFG().getBlocks();
```

Comment:
Any reason you aren't using schedule.getCFG().getNodeToBlock().get(this) instead of the linear search?

Reply:
I didn't found this function before. I will use `schedule.getCFG().blockFor(this)` instead. Thank you!